network/router.go

```diff
@@ -346,3 +356,10 @@ func (r *Router) receiveServerIdentity(c Conn) (*ServerIdentity, error) {
 	log.Lvl4(r.address, "Identity received from", dst.Address)
 	return dst, nil
 }
+
+// SetErrorHandler sets the network error handler function for this entity. The function will be called
+// on network error (e.g. TCP broken stream, host disconnected, etc) with the identity of the faulty
+// remote host as 1st parameter.
+func (r *Router) SetErrorHandler(errorHandler func(*ServerIdentity)) {
+	r.errorHappened = errorHandler
```

Comment:
Well it's `SetErrorHandler` not `AddErrorHandler`. A setter should not panic if called twice in my opinion